test/testResourceQuoteStyle.js

```diff
@@ -0,0 +1,728 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ResourceString } from 'ilib-tools-common';
+
+import ResourceQuoteStyle from "../src/rules/ResourceQuoteStyle.js";
+
+import { Result } from 'i18nlint-common';
+
+export const testResourceQuoteStyle = {
+    testResourceQuoteStyle: function(test) {
+        test.expect(1);
+
+        const rule = new ResourceQuoteStyle();
+        test.ok(rule);
+
+        test.done();
+    },
+
+    testResourceQuoteStyleName: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceQuoteStyle();
+        test.ok(rule);
+
+        test.equal(rule.getName(), "resource-quote-style");
+
+        test.done();
+    },
+
+    testResourceQuoteStyleDescription: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceQuoteStyle();
+        test.ok(rule);
+
+        test.equal(rule.getDescription(), "Ensure that the proper quote characters are used in translated resources");
+
+        test.done();
+    },
+
+    testResourceQuoteStyleSourceLocale: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceQuoteStyle({
+            sourceLocale: "de-DE"
+        });
+        test.ok(rule);
+
+        test.equal(rule.getSourceLocale(), "de-DE");
+
+        test.done();
+    },
+
+    testResourceQuoteStyleGetRuleType: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceQuoteStyle({
+            sourceLocale: "de-DE"
+        });
+        test.ok(rule);
+
+        test.equal(rule.getRuleType(), "resource");
+
+        test.done();
+    },
+
+    testResourceQuoteStyleMatchSimpleNative: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceQuoteStyle();
+        test.ok(rule);
+
+        const actual = rule.match({
+            locale: "de-DE",
+            resource: new ResourceString({
+                key: "quote.test",
+                sourceLocale: "en-US",
+                source: 'This string contains “quotes” in it.',
+                targetLocale: "de-DE",
+                target: "Diese Zeichenfolge enthält 'Anführungszeichen'.",
+                pathName: "a/b/c.xliff"
+            }),
+            file: "x"
+        });
+        // if the source contains native quotes, the target must too
+        const expected = new Result({
+            severity: "warning",
+            description: "Quote style for the the locale de-DE should be „text“",
+            id: "quote.test",
+            source: 'This string contains “quotes” in it.',
+            highlight: 'Target: Diese Zeichenfolge enthält <e0>\'</e0>Anführungszeichen<e0>\'</e0>.',
```

Comment:
IIRC highlights were supposed to have different numbers (`<e1>` etc)

Reply:
Oh yeah, good point. Let me look at that one...